src/geowombat/core/vi.py

```diff
@@ -161,24 +173,51 @@ def norm_diff_math(self, data, b1, b2, name, sensor, nodata=None, mask=False, sc
 
         band_variable = 'wavelength' if 'wavelength' in data.coords else 'band'
 
-        band_data = self.scale_and_assign(data, nodata, band_variable, scale_factor, [b1, b2], [b1, b2])
+        band_data = self.scale_and_assign(
+            data, nodata, band_variable, scale_factor, [b1, b2], [b1, b2]
+        )
 
         if band_variable == 'wavelength':
 
-            result = ((band_data.sel(wavelength=b2) - band_data.sel(wavelength=b1)) /
-                      (band_data.sel(wavelength=b2) + band_data.sel(wavelength=b1))).fillna(nodata).astype('float64')
+            result = (
+                (
+                    (band_data.sel(wavelength=b2) - band_data.sel(wavelength=b1))
+                    / (band_data.sel(wavelength=b2) + band_data.sel(wavelength=b1))
+                )
+                .fillna(nodata)
+                .astype('float64')
+            )
 
         else:
 
-            result = ((band_data.sel(band=b2) - band_data.sel(band=b1)) /
-                      (band_data.sel(band=b2) + band_data.sel(band=b1))).fillna(nodata).astype('float64')
-
-        return self.mask_and_assign(band_data, result, band_variable, b2, nodata, name, mask, -1, 1, scale_factor, sensor)
-
-    def avi_math(self, data, sensor, wavelengths, nodata=None, mask=False, scale_factor=1.0):
-
-        """
-        Advanced vegetation index
+            result = (
+                (
+                    (band_data.sel(band=b2) - band_data.sel(band=b1))
```

Comment:
First I am looking at this, so just want to check something. should this be based on band name "nir' etc instead of location?

Reply:
🤔 Are you referring to 'b1' and 'b2'? They would be band names, such as 'nir'. There is a sensor check so there is no way for this to be called without passing a sensor and a `DataArray` with wavelength band names.

Reply:
I.e., `band_data.sel(band=b2)` would be something like `band_data.sel(band='nir')`, which would be a 2d array.